docs/markdown/Contributing.md

```diff
@@ -125,7 +125,8 @@ in problematic cases.
 The tests are split into two different parts: unit tests and full
 project tests. To run all tests, execute `./run_tests.py`. Unit tests
 can be run with `./run_unittests.py` and project tests with
-`./run_project_tests.py`.
+`./run_project_tests.py`. To run a single suite you can `cd 'test cases'`
+and `../run_project_tests.py --only fortran`.
```

Comment:
You don't need to (and indeed should not) cd into the dir, the tests can be run with `./run_project_tests --only fortran`.

Also, the commit message tag should be `[skip ci]`, not `[skip_ci]`.

Reply:
> the tests can be run with ./run_project_tests --only fortran.

Sure but see shell completion rationale in the commit message. No?

>  (and indeed should not)

Why not? "works4me" :-)

> Also, the commit message tag should be [skip ci], not [skip_ci].

Apologies, will fix in v2 if any.

PS: interesting that even draft requests in progress are being looked at ;-)

Reply:
shell completion:  does not work on PowerShell / Windows. The test has to be in `all_tests` variable to be picked up.


I created the `--only` option for run_project_tests.py, because on embedded or low CPU core-count systems, the project tests can take *hours*.
However, I should have also updated these docs and done like:

```python
parser.add_argument('--only', help='name of test(s) to run', nargs='+', choices=['cmake', 'fortran', ...])
```

that would help users who enter a wrong --only choice with a command line error like
```
error: argument only: invalid choice: 'cmakeFoo' (choose from 'cmake', 'fortran', ...)
```

Reply:
> interesting that even draft requests in progress are being looked at

If you have a change that is not ready for merge, then please don't open a PR. Due to the stupid way Github shows the pull request list, draft PRs drown out ready PRs. Sadly our PR list is already a bit out of control, it would be nice to avoid clutter with non-ready draft PRs.

> Sure but see shell completion rationale in the commit message.

People don't read commit messages of documentation. They only read documentation (and often not even that, but that is unrelated). In isolation this sentence implies that you _have to be_ in the `test cases` sub dir in order to run the tests. This is not true, of course, but is the impression people get if they only read this bit.

Reply:
> If you have a change that is not ready for merge, then please don't open a PR.

Considering this goes against of [github documentation](https://github.blog/2019-02-14-introducing-draft-pull-requests/),  https://mesonbuild.com/Contributing.html should explicitly forbid draft PRs.


One typical use case for draft PRs (among others) is a code change that passes all tests locally on a subset of operating systems/toolchains. In that case a draft PR can provide missing test coverage without wasting reviewers' bandwidth thanks to the complete CI results. I've been told on IRC that there's a way to ask github/Travis to run meson's CI on a private clone, is there some documentation for this?


> Due to the stupid way Github shows the pull request list, draft PRs drown out ready PRs. Sadly our PR list is already a bit out of control, it would be nice to avoid clutter with non-ready draft PRs.

While `[-]is:draft` doesn't work (obscure conflicts with `is:open` and related), `draft:{false,true}` works perfectly. Try it.
https://github.community/t5/How-to-use-Git-and-GitHub/Pull-Request-filter-for-draft-PRs/td-p/20016

Reply:
> People don't read commit messages of documentation. 

... on github.

On the command line, the git command I use the most is "git log". Github's web interface seems to have no equivalent, I mean nothing that shows all commit messages on one page. You can only view one commit message at a time. You cannot review commit messages either. I believe this can be explained by two rationales:
- github's preferred review model is [based on fixup commits](https://github.com/zephyrproject-rtos/zephyr/pull/14444) (as opposed to force pushes)
- moving more meta data outside of plain git helps with vendor lock-in.

None of these seems like a good reason to me, so I normally keep caring about commit messages even when using github - including documentation commits because it's much easier to be consistent than to draw some lines.

Now maybe the meson project has a policy is to "go with the github flow" and not care about [some] commit messages? That would be understandable but again it would be worth a sentence or two in the github section of https://mesonbuild.com/Contributing.html because it would be a significant departure from normal git (!=github) usage.

Reply:
> In isolation this sentence implies that you have to be in the test cases sub dir in order to run the tests. This is not true, of course, but is the impression people get if they only read this bit.

Good catch, thanks. Another mistake I made was to assume (most) test suites are named after such a subdirectory. At least two reasons to abandon this PR.

Now I'm still sure whether it is _allowed_ to invoke ../run_some_tests.py from the `test cases/` directory, I mean I still wonder whether it may break in the future (seem to Just Work now). Not important.